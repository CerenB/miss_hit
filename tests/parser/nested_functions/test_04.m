% From https://www.mathworks.com/help/matlab/matlab_prog/nested-functions.html

function test_04
    nestedfun1;
    nestedfun2;

    function nestedfun1
        x = 1;
    end

    function nestedfun2
        x = 2;
    end
end
