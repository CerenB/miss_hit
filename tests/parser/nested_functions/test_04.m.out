----------------------------------------------------------------------
--  Parse tree for test_04.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <test_04>
    Body: Sequence_Of_Statements
      Statements: Naked_Expression_Statement
        Expression: Identifier <nestedfun1>
      Statements: Naked_Expression_Statement
        Expression: Identifier <nestedfun2>
    Nested: Function_Definition
      Signature: Function_Signature
        Name: Identifier <nestedfun1>
      Body: Sequence_Of_Statements
        Statements: Simple_Assignment_Statement
          LHS: Identifier <x>
          RHS: Number_Literal <1>
    Nested: Function_Definition
      Signature: Function_Signature
        Name: Identifier <nestedfun2>
      Body: Sequence_Of_Statements
        Statements: Simple_Assignment_Statement
          LHS: Identifier <x>
          RHS: Number_Literal <2>
----------------------------------------------------------------------
In test_04.m, line 4
|     nestedfun1;
|               ^ info: this token is not linked to the ast
In test_04.m, line 5
|     nestedfun2;
|               ^ info: this token is not linked to the ast
In test_04.m, line 8
|         x = 1;
|              ^ info: this token is not linked to the ast
In test_04.m, line 9
|     end
|     ^^^ info: this token is not linked to the ast
In test_04.m, line 12
|         x = 2;
|              ^ info: this token is not linked to the ast
In test_04.m, line 13
|     end
|     ^^^ info: this token is not linked to the ast
In test_04.m, line 14
| end
| ^^^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
